src/memoize.spec.js

```diff
@@ -0,0 +1,31 @@
+import memoize from './memoize';
+
+function getA(user) {
+  return user.a;
+}
+
+function getUsername(user) {
+  return user.username;
+}
+
+describe('memoize', function() {
+  it('caches values with resolver', () => {
+    const wrappedFn = jest.fn(getA);
+    const userA = {username: 'JaneDoe', a:123};
+    const userB = {username: 'Max', a:256};
+    const fn = memoize(wrappedFn, getUsername);
+    expect(fn(userA)).toBe(123);
+    expect(fn(userB)).toBe(256);
+    expect(fn(userA)).toBe(123);
```

Comment:
aren't those suppose to return JaneDoe and Max?

Reply:
No, the numbers are correct. The call to getUsername is just for the cache key.